File: Assets/zfoocs/Common/PairLS.cs
using System;
using System.Collections.Generic;

namespace zfoocs
{
    
    public class PairLS
    {
        public long key;
        public string value;

        public static PairLS ValueOf(long key, string value)
        {
            var packet = new PairLS();
            packet.key = key;
            packet.value = value;
            return packet;
        }
    }


    public class PairLSRegistration : IProtocolRegistration
    {
        public short ProtocolId()
        {
            return 113;
        }

        public void Write(ByteBuffer buffer, object packet)
        {
            if (packet == null)
            {
                buffer.WriteInt(0);
                return;
            }
            PairLS message = (PairLS) packet;
            buffer.WriteInt(-1);
            buffer.WriteLong(message.key);
            buffer.WriteString(message.value);
        }

        public object Read(ByteBuffer buffer)
        {
            int length = buffer.ReadInt();
            if (length == 0)
            {
                return null;
            }
            int beforeReadIndex = buffer.ReadOffset();
            PairLS packet = new PairLS();
            long result0 = buffer.ReadLong();
            packet.key = result0;
            string result1 = buffer.ReadString();
            packet.value = result1;
            if (length > 0) {
                buffer.SetReadOffset(beforeReadIndex + length);
            }
            return packet;
        }
    }
}

File: Assets/zfoolua/common/PairLS.lua

local PairLS = {}

function PairLS:new()
    local obj = {
        key = 0, -- long
        value = "" -- string
    }
    setmetatable(obj, self)
    self.__index = self
    return obj
end

function PairLS:protocolId()
    return 113
end

function PairLS:write(buffer, packet)
    if packet == nil then
        buffer:writeInt(0)
        return
    end
    buffer:writeInt(-1)
    buffer:writeLong(packet.key)
    buffer:writeString(packet.value)
end

function PairLS:read(buffer)
    local length = buffer:readInt()
    if length == 0 then
        return nil
    end
    local beforeReadIndex = buffer:getReadOffset()
    local packet = PairLS:new()
    local result0 = buffer:readLong()
    packet.key = result0
    local result1 = buffer:readString()
    packet.value = result1
    if length > 0 then
        buffer:setReadOffset(beforeReadIndex + length)
    end
    return packet
end

return PairLS
